docs/users/web3.mdx

```diff
@@ -0,0 +1,148 @@
+
+The following example shows you how to compose Clerk's Control Components to build authentication flows that match your needs. Please note that you don't need to use any additional APIs, everything shown below is just Javascript.
+
+```jsx filename="_app.js" copy
+import '../styles/globals.css'
+import { ClerkProvider, SignedIn, SignedOut, SignInWithMetamaskButton } from '@clerk/nextjs';
+import { useRouter } from 'next/router';
+
+//  List pages you want to be publicly accessible, or leave empty if
+//  every page requires authentication. Use this naming strategy:
+//   "/"              for pages/index.js
+//   "/foo"           for pages/foo/index.js
+//   "/foo/bar"       for pages/foo/bar.js
+//   "/foo/[...bar]"  for pages/foo/[...bar].js
+const publicPages = [];
+
+function MyApp({ Component, pageProps }) {
+  // Get the pathname
+  const { pathname } = useRouter();
+
+  // Check if the current route matches a public page
+  const isPublicPage = publicPages.includes(pathname);
+
+  // If the current route is listed as public, render it directly
+  // Otherwise, use Clerk to require authentication
+  return (
+    <ClerkProvider>
+      {isPublicPage ? (
+        <Component {...pageProps} />
+      ) : (
+        <>
+          <SignedIn>
+            <Component {...pageProps} />
+          </SignedIn>
+          <SignedOut>
+            <SignInWithMetamaskButton />
+          </SignedOut>
+        </>
+      )}
+    </ClerkProvider>
+  );
+}
+
+export default MyApp;
+```
+
+Visit http://localhost:3000 to see your page. The home "/" page is not listed in the `publicPages` array, so you'll immediately see the **Sign in with Metamask** button - go ahead and try it!
+
+## Hello, world!
+
+After signing in with Metamask, you'll be presented with the Next.js default start screen. Let's update `pages/index.js` by adding a [`<UserButton />`](/docs/components/user/user-button) to retrieve the current users' Metamask address:
+
+```jsx filename="pages/index.js" copy
+import { UserButton, useUser } from "@clerk/nextjs";
+
+export default function Home() {
+  const { primaryWeb3Wallet } = useUser();
+  return (
+    <>
+      <header
+        style={{
+          display: "flex",
+          justifyContent: "space-between",
+          padding: "1rem",
+        }}
+      >
+        <span>My first DApp</span>
+        <UserButton />
+      </header>
+      <main style={{ padding: "1rem" }}>
+        <p>Address: {primaryWeb3Wallet.web3Wallet}</p>
+      </main>
+    </>
+  );
+}
+```
+
+## Accessing the Web3 address from the backend
+
+### Add authentication to a serverless API route
+
+A new Next.js app comes with a sample API route in `/pages/api/hello.js`. Let's modify it to retrieve the user's Web3 address:
+
+```jsx filename="pages/api/hello.js" copy
+import { requireSession } from "@clerk/nextjs/api";
```

Comment:
This API is a bit older. To avoid content duplication it would be ideal to link to https://clerk.com/docs/quickstarts/nextjs#api-routes